akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -206,7 +203,10 @@ class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
                          connectTimeout: Duration = Duration.Inf,
                          idleTimeout: Duration = Duration.Inf): Flow[ByteString, ByteString, Future[OutgoingConnection]] = {
 
-    val remoteAddr = remoteAddress
+    val timeoutHandling = idleTimeout match {
+      case d: FiniteDuration ⇒ Flow[ByteString].via(Timeouts.idleTimeout(d))
```

Comment:
this should be `join(Timeouts.idleTimeoutBidi)`. Otherwise this will timeout on a long response, which is not idle at all.